### DISCUSSION TITLE
Argument of type "Literal[0]" cannot be assigned to parameter "__s" of type "slice" in function "__getitem__"

### DISCUSSION THREAD

**Discussion by devdanzin** [microsoft/pyright#5863] [Q&A] [answered]
I'm getting the error in the title when checking the code fragment below with pyright-1.1.325:

```python
metrics = [1]
v = metrics[0]
```

Is this expected behavior? Any tips on how to add hints to avoid that? Using `metrics: list[int] = [1]` seems to have no effect.

**Comment by erictraut** [comment]
I'm not able to repro your issue. The inferred type of `metrics` should be `list[int]`, and `metrics[0]` should not generate any type errors. Are you able to repro this with a clean project, a clean virtual environment, and default pyright settings? I suspect that you have something configured incorrectly. Perhaps you've specified an alternate set of `typeshed` stubs that you've modified?

**Reply by devdanzin** [reply]
No, I'm not able to repro in a clean venv with a fresh install, sorry for the noise. If I'm able to pinpoint what went wrong I'll update here so others can avoid it. 

Thanks for taking a look, sorry again.

**Comment by devdanzin** [comment]
It seems that what caused the issue for me was installing the [shed ](https://github.com/Zac-HD/shed)package, either it or one of its dependencies adds a `py.typed` file to the site-packages directory and that somehow interferes with pyright. I've narrowed it down to having that file present while [typing_extensions ](https://github.com/python/typing_extensions) is installed.

Steps to reproduce:

1. Create a new virtual environment, update pip and setuptools
2. `pip install pyright typing_extensions`
3. Create a `py.typed` file in the site packages directory (use snippet below)
4. Run pyright with `--verbose --dependencies` against the test fragment below,
5. See error as in discussion title, see that typing_extensions is somehow being imported

Snippet to create `py.typed` file:
```python
from pathlib import Path

import typing_extensions

site_packages = Path(typing_extensions.__file__).parent
py_typed = site_packages / "py.typed"
py_typed.touch(exist_ok=True)
```

Test fragment:
```python
metrics = [1]
v = metrics[0]
```

Hopefully this reproduces now, I'm on Windows if that makes a difference.

Edit to add: I've identified [autoflake](https://github.com/PyCQA/autoflake/commit/ab37b11eedfe25fd24fae9952e2ba8a9d7b928eb) as the package installing the `py.typed` file. I've created https://github.com/PyCQA/autoflake/issues/272 to see if they can avoid installing it.

**Reply by erictraut** [reply, marked as answer]
Yikes, it's really bad that someone is installing a `py.typed` file in `site-packages`. That's going to cause all libraries installed in `site-packages` to be interpreted as though they are typed. They'll even override typeshed stubs that ship with pyright, such as the stub file for `typing_extensions.pyi`.

Please file a bug against the package that is doing this!
